mlflow/exceptions.py

```diff
@@ -38,9 +38,9 @@ class RestException(MlflowException):
     """Exception thrown on non 200-level responses from the REST API"""
     def __init__(self, json):
         error_code = json.get('error_code', INTERNAL_ERROR)
-        message = error_code
-        if 'message' in json:
-            message = "%s: %s" % (error_code, json['message'])
+        message = "%s: %s" % (error_code, json['message']) if 'message' in json else \
```

Comment:
1. Can we include json in the message here? Wouldn't that disrupt json serialization of the exception?
2.  I wonder if we need to include error_code in the message here since the parent already includes  human readable error code in the output.

Reply:
1. I replaced it with "Response: " + str(json) to avoid passing the dict and maintain the fact that it is a message string as is done for non json responses.
2, I removed it. The json serialization preserves it but now the exception repr does not include the int

Reply:
I think the inclusion is odd but there are tests verifying it exists(in test_rest_utils.py).
I reverted its removal since I could not verify there were no external dependencies on the shape of the message.

Reply:
makes sense.